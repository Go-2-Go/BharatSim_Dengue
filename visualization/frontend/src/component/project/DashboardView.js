import React from 'react';
import Box from '@material-ui/core/Box';

function TabPanel(props) {
  const { children, value, index } = props;

  return <div>{value === index && <Box p={3}>{children}</Box>}</div>;
}

export default function DashboardView({ views, value }) {
  return (
    <Box>
      {views.map((view, index) => (
        <TabPanel value={value} index={index} key={`value-${view.name}`}>
          {view.name}
        </TabPanel>
      ))}
    </Box>
  );
}
