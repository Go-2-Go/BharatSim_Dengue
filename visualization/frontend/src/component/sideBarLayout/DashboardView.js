import React from 'react';
import Box from '@material-ui/core/Box';
import TabPanel from '../../uiComponent/TabPanel';
import ConfigureDashboardData from '../project/ConfigureDashboardData';

export default function DashboardView({ projectName, dashboardData, value }) {
  return (
    <Box>
      {dashboardData.map((dashboard, index) => (
        <TabPanel value={value} index={index} key={`value-${dashboard.name}`}>
          <ConfigureDashboardData dashboardData={dashboard} projectName={projectName} />
        </TabPanel>
      ))}
    </Box>
  );
}
