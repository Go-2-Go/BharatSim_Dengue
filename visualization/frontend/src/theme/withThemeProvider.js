import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme() {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
