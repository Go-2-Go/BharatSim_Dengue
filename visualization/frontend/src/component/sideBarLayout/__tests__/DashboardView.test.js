import React from 'react';
import { render } from '@testing-library/react';
import DashboardView from '../DashboardView';
import withThemeProvider from '../../../theme/withThemeProvider';

describe('Dashboard View', function () {
  const DashboardViewWithTheme = withThemeProvider(DashboardView);
  it('should render dashboard view for given dashboard data and project name ', function () {
    const { container } = render(
      <DashboardViewWithTheme
        value={0}
        dashboardData={[{ name: 'dashboard1' }, { name: 'dashboard2' }]}
        projectName="project1"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
