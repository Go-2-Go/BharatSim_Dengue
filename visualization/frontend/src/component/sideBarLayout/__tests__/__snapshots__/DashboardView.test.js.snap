// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard View should render dashboard view for given dashboard data and project name  1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      aria-labelledby="simple-tab-0"
      id="simple-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <div
          class="MuiBox-root MuiBox-root-7 makeStyles-configureProjectDataBar-2"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
             Configure Dashboard Data
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
               Back to recent projects
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-8 makeStyles-dashboardDataContainer-4"
        >
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
               Manage Dashboard Dataset
            </h6>
          </div>
          <div
            class="MuiBox-root MuiBox-root-10 makeStyles-dashboardDataHeader-3"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              project1 :: dashboard1
            </h6>
          </div>
          <div
            class="MuiBox-root MuiBox-root-11 makeStyles-dashboardDataBody-5"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
            >
              Before we can create any visualization, we â€˜ll need some data.
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Use Upload dataset to add data files to your dashboard.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-1"
      hidden=""
      id="simple-tabpanel-1"
      role="tabpanel"
    />
  </div>
</div>
`;
